Amity Develop update

ianoti opened the pull request:
the remaining functionality to implement is:
`add database persistence`
`implement doc opt frontend`
`increase test coverage`

ianoti committed:
initial commit

ianoti committed:
Amity initial commit of UML class diagram

ianoti committed:
added option to ignore .DS_Store

ianoti committed:
(Chore) minor folder reorganisation

ianoti committed:
minor fix to file to ignore all .DS_Store files

ianoti committed:
remodelled class attributes

ianoti committed:
added relative imports for test modules

ianoti committed:
fixed incorrect inheritance of class attributes

ianoti committed:
updated attribute definition for room class

ianoti committed:
worked on tests for amity

modified test for room creation and modified access attribute datatype

fixed incorrect operation order for object creation

fixed incorrect assertion code

reworked access level test to make more comprehensive

rewrote accomodation test

ianoti committed:
reorganised project folder

ianoti committed:
fixed import bug for the model module

ianoti committed:
initial commit of Amity interface, class based implementation

ianoti committed:
initial commit of amity module tests

ianoti committed:
refactored code to treat folders as packages for import

ianoti committed:
edited model to add occupant attribute to room class

ianoti committed:
added static methods to amity class

ianoti committed:
added occupant attribute to room model

ianoti committed:
added test for room creation method in Amity

ianoti committed:
added dummy functionality for load person and add person

ianoti committed:
added dummy methods to class for unit testing

ianoti committed:
edited to exclude test data and files

ianoti committed:
test data used in tests

ianoti committed:
fixed shebang and add_room() variable order

ianoti committed:
added tests for add person method and batch adding of persons

fixed bug of incorrect implementation of tearDown method

added edge case test cases for adding rooms and people to amity system

ianoti committed:
add /data folder for ease of loading batch .txt files for name addition

ianoti committed:
added more names to test data for room allocation test

added test setup to test allocation function of system

ianoti committed:
(feature) worked on features for add_room and add_person functions

refactored add room code to only process string variables and accept multiple inputs

added optional arguments to add_person function

minor edits of test_add_batchpeople, test_room_add_wrong_options, test_add_person_wrong_options

ianoti committed:
renamed person_list to waiting_list

person_list renamed again

ianoti committed:
(feature) added functionality to create rooms

ianoti committed:
(chore) reworked erroneous input test to reduce ambiguity and room creation test for correct assertions

ianoti committed:
(feature) system can add single users and catch user errors

ianoti committed:
(chore) minor fix to assertions to test for room occupancy

ianoti committed:
(feature) added system can add persons from text file

ianoti committed:
initial commit to develop branch

ianoti committed:
(feature) added alllocation functionality to Amity

ianoti committed:
(bug fix) and (feature) remodelled room class to fix sublist bugs in Amity and added features

(feature) added functionality to print unallocated and allocated persons

(feature) added reallocation functionality and fixed allocation bug added print room functionality

(feature) added person id attribute to model

ianoti committed:
(feature) added feature to print out allocations and unallocated to file

ianoti committed:
modified .gitignore

ianoti committed:
refactored tests

ianoti committed:
(feature)added docopt skeleton for amity

(feature)added docopt skeleton for amity

(feature) added methods to docopts framework

ianoti committed:
(chore) reworked code for improved compatibility with docopt framework

(feature) added functionality of saving system state to database and loading state from database

(chore) added attribute to models to return type

tweaked logic for databse loading by adding encoding

njirap commented on app.py:
```diff
@@ -0,0 +1,284 @@
+                return "the role is invalid"
+            self.waiting_list.append(person)
+            self.people_directory.append(person)
+        elif not isinstance(fname, str) or not isinstance(sname, str):
```
this is just sufficient being `else` and not an `elif`

njirap commented on app.py:
```diff
@@ -0,0 +1,284 @@
+            return "the name is invalid"
+
+    def add_room(self, rm_type, given_names):
+        """ method to add rooms to system utilising models specified """
```
```
if type(rm_name) is not str:
    return "the room name is invalid"
if type(rm_type) is not str:
    return "the room option is invalid"
```

ianoti committed:
modified .gitignore to eliminate unnecessary files

ianoti committed:
(chore) formatted code for PEP8 compliance

ianoti committed:
(chore) modified code to add functionality and interface

(chore) refactored tests

(chore) refactored code for PEP8 and improved error handling

(chore) refactored docopt file to return program statements

(chore) refactored code and added error handling for databases and files

(chore) minor refactoring of docopt file

ianoti committed:
added requirements.txt file

ianoti committed:
(feature) added test for the database functionality

-(chore) refactored reallocate function to work after database loading

ianoti committed:
modified .gitignore

ianoti committed:
refactored tests

ianoti committed:
(chore) refactored get_id function to allow for multiple users to be displayed if the name is the same

ianoti committed:
(chore) minor refactoring of tests

made some minor changes to tests, removed unnecessary asserts

ianoti commented:
I have modified the system and done the changes. I also refactored the `print allocations` and `print unallocated` functions to reduce repetition.

ianoti committed:
updated README

ianoti committed:
(chore) removed unused line of code
